var/spack/repos/builtin/packages/py-pynio/package.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyPynio(PythonPackage):
+    """PyNIO ("pie-nee-oh") is a Python module that allows read and/or
+       write access to a variety of scientific data formats popular in
+       climate and weather"""
+
+    homepage = "https://www.pyngl.ucar.edu/Nio.shtml"
+    url      = "https://github.com/NCAR/pynio/archive/1.5.4.tar.gz"
+
+    version('1.5.4', sha256='e5bb57d902740d25e4781a9f89e888149f55f2ffe60f9a5ad71069f017c89e1a')
+
+    depends_on('netcdf')
+    depends_on('libpng')
+    depends_on('py-numpy', type=('build', 'run'))
```

Comment:
Looks like there are a lot of other optional dependencies, like HDF and GDAL. If you don't need them, I won't make you add them, but I'm sure someone will want them someday.

Reply:
No, I will add them. You are always much better at noticing dependencies than I am. I built this for someone else, but for sure this will come up. Now that you have said it, I can see what you saw in the setup.py. I wish was smarter....

Reply:
Note to self - just because it builds, doesn't mean it is right..... sigh.....

Reply:
No worries. Yeah, I pretty much only look at the `setup.py`. Developers are terrible at actually listing their dependencies. I'm not sure if it helps at all, but I listed a lot of tips on locating Python dependencies in https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html.